File: Dataset/Leetcode/test/55/442.java
 class Solution {
    public boolean XXX(int[] nums) {
        int[] shawdow = new int[nums.length];
        shawdow[0] = 1;
        for (int i = 0; i < nums.length; i++) {
            // 当前点未被标记，证明不可达
            if (shawdow[i] == 0) {
                return false;
            }
            // 依次标记当前点可以跳达的地方
            for (int k = nums[i] + i; k >= i; k--) {
                if (k >= nums.length - 1) {
                    return true;
                }
                if (shawdow[k] == 1) {
                    break;
                }
                shawdow[k] = 1;
            }
        }
        return false;
    }
}

File: Dataset/Leetcode/train/26/442.cpp
 class Solution {
public:
    void swap(vector<int>& nums,int l,int r){
        int temp=nums[l];
        nums[l]=nums[r];
        nums[r]=temp;
    }
    int XXX(vector<int>& nums) {
        if(nums.size()<2)return nums.size();
        int l=0,r=1;
        while(r<nums.size()){
            if(nums[l]==nums[r])r++;
            else{
                swap(nums,++l,r++);
            }
        }
        return ++l;
    }
};

//假如修改原数组
class Solution {
public:
    int XXX(vector<int>& nums) {
        if(nums.size()<2)return nums.size();
        int l=0,r=1;
        while(r<nums.size()){
            if(nums[l]==nums[r])r++;
            else nums[++l]=nums[r++];
        }
        return ++l;
    }
};

